### PR TITLE
feature/SWTCH-1112 separate connection to cache server and did registry from iam ininitialization

### PR DESCRIPTION
**Pull request by JGiter**
- Connecting to cache server and DID registry are moved to `connectToCacheServer` and `connectToDID`. These methods can be invoked after iam initialization or as part of it by setting additional initialization flags `initCacheServer` and `initDID`

### CODE REVIEW

File: src/iam/iam-base.ts [hunk 1]
```diff
@@ -413,8 +428,8 @@ export class IAMBase {
   }
 
   private setJWT() {
-    if (this._didSigner) {
-      this._jwt = new JWT(this._didSigner);
+    if (this._signer) {
+      this._jwt = new JWT(this._signer);
```
Comment (jrhender) [inline src/iam/iam-base.ts:432]: Nice 👍

**Review by jrhender — APPROVED** [review]
This looks great, nice work 👍

**Review by DawidGil-jlabs — APPROVED** [review]
Lgtm. What i understood i can leave initialization for switchboard as it is, and for patron stake view i just need to put `{ initCacheServer: false }` and that's all.

**Comment by jrhender** [comment]
> Lgtm. What i understood i can leave initialization for switchboard as it is, and for patron stake view i just need to put `{ initCacheServer: false }` and that's all.

Yes, I think that is basically right. I think you put `{ walletProvider: Metamask, initCacheServer: false, initDid: false }` for the patron stake view

**Comment by DawidGil-jlabs** [comment]
> > Lgtm. What i understood i can leave initialization for switchboard as it is, and for patron stake view i just need to put `{ initCacheServer: false }` and that's all.
> 
> Yes, I think that is basically right. I think you put `{ walletProvider: Metamask, initCacheServer: false, initDid: false }` for the patron stake view

Oh ok. I thought `initDid` should be enabled, but i do not have knowledge how this initialization works. Thanks for clarification :)

**Pull request merged**